pkg/engine/builtinruntime.go

```diff
@@ -16,15 +16,16 @@ package engine
 
 import (
 	"context"
+	"fmt"
 	"io"
 
-	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/apply-replacements/replacements"
-	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/set-namespace/transformer"
-	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/starlark/starlark"
 	fnsdk "github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
 	"github.com/nephio-project/porch/pkg/kpt"
 	v1 "github.com/nephio-project/porch/pkg/kpt/api/kptfile/v1"
 	"github.com/nephio-project/porch/pkg/kpt/fn"
+	"github.com/nephio-project/porch/pkg/kpt/fn/third_party/GoogleContainerTools/kpt-functions-catalog/functions/go/apply_replacements"
```

Comment:
I understand the logic in pulling these in but it concerns me that we then "own" them.
Not quite clear on what these are used for either. Could you explain?

Reply:
sure. the problem with KRM (formerly kpt) functions is that actually calling out to function-runner to execute the docker container and swnd back the image can be ridiculously slow for simply things like replacing a namespace. Because of this there are various spped up tricks are buolt into porch to make calling at least know/core functions faster (e.g. the binary of lot of them is actually copied to the function-runner container to avoid actually starting a new pod). One of those speed up measured is to have a so-called "built-in executor" in porch that instead of relying on function-runner calls the golang function for some known KRM functions. 
This require us to import the go code into porch. The problem with this  (as you and me are painfully aware) is that the code of those functions still uses outdated versions of the kpt-fn-sdk that is incompatible with our own PackageVariant code. That's why instead of simply importing it I copied/forked the code from the catalog repo, updated the code to use the latest version of the sdk. And all of this let's us get rid of the dependency on the ancient sdk version. yay :)

Please note that in the future we can think of other solutions, e.g. we can fork the whole catalog, since the old one seems yo be abandoned by Google.

(I also fixed a bug that in case of a panicking KRM function, that is normal behavior on invalid input btw., also panics the whole porch server)

Reply:
Much clearer now. Thanks